frontend/packages/console-app/src/components/detect-namespace/namespace.ts

```diff
@@ -0,0 +1,51 @@
+import * as React from 'react';
+// eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+// @ts-ignore: FIXME missing exports due to out-of-sync @types/react-redux version
+import { useDispatch } from 'react-redux';
+import { useLocation } from 'react-router-dom';
+import { getNamespace } from '@console/internal/components/utils/link';
+import { useUserSettingsCompatibility } from '@console/shared/src/hooks/useUserSettingsCompatibility';
+import { setActiveNamespace } from '@console/internal/actions/ui';
+import { ALL_NAMESPACES_KEY } from '@console/shared/src/constants';
+
+type NamespaceContextType = {
+  namespace?: string;
+  setNamespace?: (ns: string) => void;
+};
+
+const LAST_NAMESPACE_NAME_LOCAL_STORAGE_KEY = `bridge/last-namespace-name`;
+const LAST_NAMESPACE_NAME_USER_SETTINGS_KEY = 'console.lastNamespace';
+
+export const NamespaceContext = React.createContext<NamespaceContextType>({});
+
+export const useValuesForNamespaceContext = () => {
+  const { pathname } = useLocation();
+  const urlNamespace = getNamespace(pathname);
+  /**
+   * [TODO]: use favorite namespace here if there is any
+   *  const [favorite] = useUserSettings()
+   */
+  const [namespace, setNs, loaded] = useUserSettingsCompatibility(
+    LAST_NAMESPACE_NAME_USER_SETTINGS_KEY,
+    LAST_NAMESPACE_NAME_LOCAL_STORAGE_KEY,
+    '',
+  );
+  const dispatchreduxAction = useDispatch();
+  const setNamespace = React.useCallback(
+    (ns: string) => {
+      dispatchreduxAction(setActiveNamespace(ns));
```

Comment:
If I understand this correctly, we're putting `activeNamespace` in a top level context provider. And also in redux? if it's already in top level context do we need to keep it in redux store as well? I am confused because i see in some places we're using value from redux and in other places using value directly from context.

Reply:
Initial idea was that only, but in some of the utils getActiveNamespace is being called which access the store directly `store.get(...)`. Didn't want to refactor everything on the last day 😞 but Yeah I agree with you using the context data everywhere and not use redux state.

Reply:
I am okay with mixed usage for now since it's the last day of feature freeze. We should definitely look at removing all usage of redux state later.

Reply:
@sahil143 please log a ticket for this so it does not get lost.